core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/buffer.go

```diff
@@ -1,13 +1,16 @@
 package logprovider
 
 import (
+	"encoding/hex"
 	"math/big"
 	"sort"
 	"sync"
 	"sync/atomic"
 
 	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
 	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/ocr2keepers/pkg/v3/random"
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg/v3/types"
```

Comment:
not fully related to this PR, but can we clean up the variables in line 17-20

allowedLogsPerBlock = 128 -> maxLogsPerUpkeepPerBlock, set it to something like 10?
bufferMaxBlockSize = 1024 -> maxLogsPerBlock

Reply:
I've set `maxLogsPerUpkeepInBlock` to 32